libraries/I2S/src/I2S.h

```diff
@@ -0,0 +1,86 @@
+/*
+  Based off of ArduinoCore-SAMD I2S interface.  Modified for the
+  ESP8266 by Earle F. Philhower, III <earlephilhower@yahoo.com>
+
+  Copyright (c) 2016 Arduino LLC.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _I2S_H_INCLUDED
+#define _I2S_H_INCLUDED
+
+#include <Arduino.h>
+#include <core_esp8266_i2s.h>
+
+typedef enum {
+  I2S_PHILIPS_MODE // Only mode allowed for now by the core
+} i2s_mode_t;
+
+class I2SClass : public Stream
+{
+public:
+  // By default only transmit and drive the clock pins
+  I2SClass(bool enableTransmit = true, bool enableRecv = false, bool driveClocks = true);
+
+  // Only 16 and 24 bps are allowed by the hardware
+  int begin(i2s_mode_t mode, long sampleRate, int bitsPerSample);
+  void end();
+
+  // from Stream
+  virtual int available();
+  virtual int read();
+  virtual int peek();
+  virtual void flush();
+
+  // from Print
+  virtual size_t write(uint8_t);
+  virtual size_t write(const uint8_t *buffer, size_t size);
+
+  virtual int availableForWrite();
+
+  int read(void* buffer, size_t size);
+
+  size_t write(int32_t);
+  size_t write(const void *buffer, size_t size);
```

Comment:
I would add a comment specifying that `::write(sample)` is blocking but `::write(buffer,len)` is not.